File: pet-store-service/src/main/java/com/chewy/osn/factory/ObjectMapperBeanEventListener.java
package com.chewy.osn.factory;

import com.chewy.osn.ObjectMapperBuilder;
import com.fasterxml.jackson.databind.ObjectMapper;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import jakarta.inject.Singleton;

@Singleton
public class ObjectMapperBeanEventListener implements BeanCreatedEventListener<ObjectMapper> {

    @Override
    public ObjectMapper onCreated(BeanCreatedEvent<ObjectMapper> event) {
        if (event.getBeanIdentifier().getName().equals("yaml")) {
            return event.getBean();
        }
        return new ObjectMapperBuilder().build();
    }
}

File: pet-store-service/src/main/kotlin/com/johnkruss/osn/factory/ObjectMapperBeanEventListener.kt
package com.johnkruss.osn.factory

import com.fasterxml.jackson.databind.ObjectMapper
import com.johnkruss.osn.ObjectMapperBuilder
import io.micronaut.context.event.BeanCreatedEvent
import io.micronaut.context.event.BeanCreatedEventListener
import io.micronaut.core.annotation.NonNull
import jakarta.inject.Singleton

@Singleton
class ObjectMapperBeanEventListener : BeanCreatedEventListener<ObjectMapper> {
    override fun onCreated(event: @NonNull BeanCreatedEvent<ObjectMapper>): ObjectMapper {
        return ObjectMapperBuilder().build()
    }
}
